compiler/lexer.nim

```diff
@@ -386,82 +402,102 @@ proc getNumber(L: var Lexer, result: var Token) =
       discard matchUnderscoreChars(L, result, {'0'..'9'})
     if L.buf[L.bufpos] in {'e', 'E'}:
       result.tokType = tkFloatLit
-      eatChar(L, result, 'e')
+      eatChar(L, result, L.buf[L.bufpos])
       if L.buf[L.bufpos] in {'+', '-'}:
         eatChar(L, result)
       discard matchUnderscoreChars(L, result, {'0'..'9'})
   endpos = L.bufpos
 
   # Second stage, find out if there's a datatype suffix and handle it
   var postPos = endpos
-  if L.buf[postPos] in {'\'', 'f', 'F', 'd', 'D', 'i', 'I', 'u', 'U'}:
-    if L.buf[postPos] == '\'':
-      inc(postPos)
-
+  var hasSuffix = false
+  var internalSuffix = false
+  if L.buf[postPos] == '\'':
+    hasSuffix = true
+    inc(postPos)
+  # 2A: handle the builtin literal versions
+  if L.buf[postPos] in {'f', 'F', 'd', 'D', 'i', 'I', 'u', 'U'}:
     case L.buf[postPos]
     of 'f', 'F':
-      inc(postPos)
-      if (L.buf[postPos] == '3') and (L.buf[postPos + 1] == '2'):
+      if (L.buf[postPos + 1] == '3') and (L.buf[postPos + 2] == '2'):
```

Comment:
how about this in followup PR?

`if (L.buf.continuesWith(postPos + 1, "32"))` would be more readable

(introducing a `func continuesWith(a: openArray[char], start: int, b: string)`)

(and could be reused in cases below)

Reply:
I've simplified it using a `cmpSuffix` proc. While not inclusive as your template, it has the benefit of not asking the future programmer to remember what the template does. There are always pros/cons to everything of course.